### PR TITLE
Fix: ColorRange incompatible type cast

### PR DESCRIPTION
**Pull request by Benglin**
### Purpose

This pull request is meant to fix a crash in `ColorRange` node when it receives invalid input of type other than `DSCore.Color`. The defect is being tracked internally as:

- [MAGN-7319](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7319) Color Range node crashes Dynamo when there is type mismatch in arguments

The following code assume every element in the collection is of type `DSCore.Color`:

```csharp
colors.AddRange(data.GetElements().Select(e => e.Data).Cast<Color>());
```

It should instead be done this way to handle possibly unknown object types:

```csharp
colors.AddRange(data.GetElements().Select(e => e.Data).OfType<Color>());
```

A new test case `InvalidInputShouldNotCrashColorRangeNode` is added to verify this fix (the test fails before the fix and pass after it).

### Declarations

Check these iff you believe they are true

- [x] The code base is in a better state after this PR
- [x] The level of testing this PR includes is appropriate

### Reviewers

Hi @junmendoza, as discussed, please take a look, thanks!

### CODE REVIEW

File: src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/ColorRange.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ public void CustomizeView(ColorRange model, NodeView nodeView)
                         var data = colorsMirror.GetData();
                         if (data.IsCollection)
                         {
-                            colors.AddRange(data.GetElements().Select(e => e.Data).Cast<Color>());
+                            colors.AddRange(data.GetElements().Select(e => e.Data).OfType<Color>());
```
Comment (Benglin) [inline src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/ColorRange.cs]: `data.GetElements` returned something that is not `Color` type, so hard casting it to `Color` type will crash.

Comment (junmendoza) [inline src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/ColorRange.cs]: I assume OfType should be just be filtering all Color types.
If so then looks good

**Comment by junmendoza** [comment]
LGTM

**Comment by Benglin** [comment]
Thanks pal!

**Pull request merged**